### PR TITLE
client: fix compile warning

### PR DESCRIPTION
**Pull request by lxbsz**
warning: suggest parentheses around ‘-’ inside ‘<<’ [-Wparentheses]
   max_fwd = 1 << (max_fwd * CHAR_BIT) - 1;
                  ~~~~~~~~~~~~~~~~~~~~~^~~

Fixes: https://tracker.ceph.com/issues/55409
Reported-by: Jos Collin <jcollin@redhat.com>
Reported-by: Rishabh Dave <ridave@redhat.com>
Signed-off-by: Xiubo Li <xiubli@redhat.com>





<!--
  - Please give your pull request a title like

      [component]: [short description]

  - Please use this format for each git commit message:

      [component]: [short description]

      [A longer multiline description]

      Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
      Signed-off-by: [Your Name] <[your email]>

    For examples, use "git log".

  - The Signed-off-by line in every git commit is important; see <span class="x x-first x-last">[Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/</span>SubmittingPatches.rst<span class="x x-first x-last">)</span>
-->

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by joscollin — APPROVED** [review]

File: src/client/Client.cc [hunk 1]
```diff
@@ -2552,7 +2552,7 @@ void Client::handle_client_request_forward(const MConstRef<MClientRequestForward
    * using the hardcode here.
    */
   int max_fwd = sizeof(((struct ceph_mds_request_head*)0)->num_fwd);
-  max_fwd = 1 << (max_fwd * CHAR_BIT) - 1;
+  max_fwd = (1 << (max_fwd * CHAR_BIT)) - 1;
```
Comment (rishabh-d-dave) [inline src/client/Client.cc:2555]: The expression `1 << (max_fwd * CHAR_BIT) - 1` is same as `1 << ((max_fwd * CHAR_BIT) - 1)` and not same as `(1 << (max_fwd * CHAR_BIT)) - 1`. After discussion with Xiubo, I found that the desired value is 255. So the first two are incorrect and last one is correct.

I've created a tracker ticket for this bug. @lxbsz Can you please include it? Here's the link - https://tracker.ceph.com/issues/55409

**Review by rishabh-d-dave — CHANGES_REQUESTED** [review]

File: src/client/Client.cc (hunk 1 above)
Comment (lxbsz) [reply src/client/Client.cc:2555]: Thanks @rishabh-d-dave very much for this. I have updated it.

Comment (lxbsz) [reply src/client/Client.cc:2555]: @joscollin Update this PR and nothing changed in the code, just add more in the commit comments and add the tracker link to the commit.

Comment (joscollin) [reply src/client/Client.cc:2555]: > And also the '_' has precedence over the '<<', more detail please

It's '-'

Comment (lxbsz) [reply src/client/Client.cc:2555]: Done.

Comment (joscollin) [reply src/client/Client.cc:2555]: Looks good, already approved.

**Review by rishabh-d-dave — APPROVED** [review]

**Comment by vshankar** [comment]
jenkins test api

**Comment by vshankar** [comment]
jenkins test make check

**Review by dparmar18 — APPROVED** [review]

**Review by nmshelke — APPROVED** [review]

**Comment by joscollin** [comment]
jenkins test make check

**Review by vshankar — APPROVED** [review]

**Pull request merged**